packages/ethers/src/client.ts

```diff
@@ -341,6 +343,32 @@ export class Web3Modal extends Web3ModalScaffold {
         const txReceipt = await txResponse.wait()
 
         return (txReceipt?.hash as `0x${string}`) || null
+      },
+
+      writeContract: async (data: WriteContractArgs) => {
+        const chainId = EthersStoreUtil.state.chainId
+        const provider = EthersStoreUtil.state.provider
+        const address = EthersStoreUtil.state.address
+
+        if (!provider) {
+          throw new Error('connectionControllerClient:sendTransaction - provider is undefined')
+        }
+
+        if (!address) {
+          throw new Error('connectionControllerClient:sendTransaction - address is undefined')
+        }
+
+        const browserProvider = new BrowserProvider(provider, chainId)
+        const signer = new JsonRpcSigner(browserProvider, address)
+        const contract = new Contract(data.tokenAddress, erc20ABI, signer)
+
+        if (!contract || typeof contract['transfer'] !== 'function') {
+          throw new Error('Contract or transfer method is undefined')
+        }
+
+        const tx = await contract['transfer'](data.receiverAddress, data.tokenAmount)
```

Comment:
Is this really needed? I'd think  `writeContract` would be generic, not tied to `transfer`. 
Any way we can make this method to execute any contract write and not just transfers?